### PR TITLE
Vacuum mask eels

### PR DESCRIPTION
**Pull request by ericpre**
Similarly to the way it can be currently done with `EDSSpectrum`, this PR add a `vacuum_mask` method to the `EELSSpectrum` signal.

### Progress of the PR
- [x] Add `vacuum_mask` method to `EELSSpectrum`,
- [x] make the usage of mask more consistent across different method by accepting signal of bool or array of bool,
- [x] fix bug with `navigation_mask` in the `quantification` method of EDS signal, which was ignored.
- [x] update docstring (if appropriate),
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
ll = hs.datasets.artificial_data.get_low_loss_eels_line_scan_signal()*100
ll.inav[:4] *= 0
ll.vacuum_mask().plot()
```

### CODE REVIEW

File: hyperspy/_signals/eds_tem.py [hunk 1]
```diff
@@ -437,6 +437,11 @@ def vacuum_mask(self, threshold=1.0, closing=True, opening=False):
         opnening: bool
             If true, applied a morphologic opening to the mask
 
+        Return
```
Comment (dnjohnstone) [inline hyperspy/_signals/eds_tem.py]: Should probably be Returns rather than Return

File: hyperspy/_signals/eels.py [hunk 2]
```diff
@@ -1365,6 +1367,52 @@ def rebin(self, new_shape=None, scale=None, crop=True, out=None):
...
+        threshold: float
+            For a given pixel, mean value in the energy axis below which the
+            pixel is considered as vacuum.
+        start_energy: float, None
+            Energy from which the intensity is averaged up to the end.
+            If None, set the value to 15.0 eV when the zero-loss peak and the
+            15.0 eV energy channel are in the spectrum, otherwise, take the 
+            last quarter of the spectrum.
+        closing: bool
+            If true, applied a morphologic closing to the mask
```
Comment (dnjohnstone) [inline hyperspy/_signals/eels.py]: If True, a morphological closing is applied to the mask.

File: hyperspy/_signals/eels.py [hunk 3]
```diff
@@ -1365,6 +1367,52 @@ def rebin(self, new_shape=None, scale=None, crop=True, out=None):
...
+        opnening: bool
+            If true, applied a morphologic opening to the mask
```
Comment (dnjohnstone) [inline hyperspy/_signals/eels.py]: If True, a morphologlical opening is applied to the mask.

File: hyperspy/_signals/eels.py [hunk 4]
```diff
@@ -1365,6 +1367,52 @@ def rebin(self, new_shape=None, scale=None, crop=True, out=None):
...
+            For a given pixel, mean value in the energy axis below which the
+            pixel is considered as vacuum.
+        start_energy: float, None
+            Energy from which the intensity is averaged up to the end.
+            If None, set the value to 15.0 eV when the zero-loss peak and the
+            15.0 eV energy channel are in the spectrum, otherwise, take the 
+            last quarter of the spectrum.
+        closing: bool
+            If true, applied a morphologic closing to the mask
+        opnening: bool
```
Comment (dnjohnstone) [inline hyperspy/_signals/eels.py]: Typo --> opening

File: hyperspy/_signals/eels.py [hunk 5]
```diff
@@ -1365,6 +1367,52 @@ def rebin(self, new_shape=None, scale=None, crop=True, out=None):
...
 
+    def vacuum_mask(self, threshold=10.0, start_energy=None,
+                    closing=True, opening=False):
+        """
+        Generate mask of the vacuum region
+
+        Parameters
+        ----------
+        threshold: float
+            For a given pixel, mean value in the energy axis below which the
```
Comment (dnjohnstone) [inline hyperspy/_signals/eels.py]: Perhaps "For a given navigation coordinate, the mean value in the energy axis..."

File: hyperspy/_signals/eels.py [hunk 6]
```diff
@@ -1365,6 +1367,52 @@ def rebin(self, new_shape=None, scale=None, crop=True, out=None):
...
+            pixel is considered as vacuum.
+        start_energy: float, None
+            Energy from which the intensity is averaged up to the end.
```
Comment (dnjohnstone) [inline hyperspy/_signals/eels.py]: Minimum energy included in the calculation of the mean intensity.

File: hyperspy/_signals/eels.py [hunk 7]
```diff
@@ -1365,6 +1367,52 @@ def rebin(self, new_shape=None, scale=None, crop=True, out=None):
...
+            last quarter of the spectrum.
+        closing: bool
+            If true, applied a morphologic closing to the mask
+        opnening: bool
+            If true, applied a morphologic opening to the mask
+
+        Return
+        ------
+        mask: signal
+            The mask of the region
```
Comment (dnjohnstone) [inline hyperspy/_signals/eels.py]: fullstop at end.

File: hyperspy/_signals/eels.py [hunk 8]
```diff
@@ -788,21 +790,21 @@ def fourier_ratio_deconvolution(self, ll,
                 'after_fourier_ratio_deconvolution')
         return cl
 
-    def richardson_lucy_deconvolution(self, psf, iterations=15, mask=None,
+    def richardson_lucy_deconvolution(self, psf, iterations=15,
```
Comment (dnjohnstone) [inline hyperspy/_signals/eels.py:1010]: Does this create a change in useage that needs documentation - or is it just a spurious kwarg that was actually not used properly in prior code?

**Review by dnjohnstone — CHANGES_REQUESTED** [review]
Seems to work for me insofar as I understand it should from the tests provided.

It seems to me that for consistency there should be a super().decomposition() for EELSSpectrum in the same way that there is for EDSTEMSpectrum with a flag to use this method to determine a vacuum mask for decomposition as default.

Other than this, and minor typos - looks good to me

**Comment by dnjohnstone** [comment]
@ericpre can you deal with the very minor comments I had so that this can be merged?

File: hyperspy/learn/mva.py [hunk 9]
```diff
@@ -201,6 +203,9 @@ def decomposition(self,
         # Transform the data in a line spectrum
         self._unfolded4decomposition = self.unfold()
         try:
+            if isinstance(navigation_mask, BaseSignal):
+                self._check_navigation_mask(mask=navigation_mask)
+                navigation_mask = navigation_mask.data
```
Comment (tjof2) [inline hyperspy/learn/mva.py:365]: This is probably a conflict with the recent decomposition overhaul

**Review by tjof2 — COMMENTED** [review]
Aside from the comments from @dnjohnstone , there's a couple of merge conflicts.

**Comment by tjof2** [comment]
Requires the merge conflicts to be resolved plus a few of the above comments

**Review by tjof2 — APPROVED** [review]

**Comment by tjof2** [comment]
Is that everything now @ericpre?

**Comment by ericpre** [comment]
Yes, this is good now. Increasing coverage made me find a bug with the navigation mask of the `quantification` method!

**Pull request merged**